enterprise/kernel/src/main/java/org/neo4j/kernel/enterprise/builtinprocs/BuiltInProcedures.java

```diff
@@ -99,7 +98,13 @@ public void setTXMetaData( @Name( value = "data" ) Map<String,Object> data )
                             "keys and values to be less than %d, got %d", HARD_CHAR_LIMIT, totalCharSize ) );
         }
 
-        tx.acquireStatement().queryRegistration().setMetaData( data );
+        getCurrentTx().acquireStatement().queryRegistration().setMetaData( data );
+    }
+
+    private KernelTransaction getCurrentTx()
+    {
+        return graph.getDependencyResolver().resolveDependency( ThreadToStatementContextBridge.class )
+                .getKernelTransactionBoundToThisThread( true );
```

Comment:
Is there a specific reason as to why this is preferred to injecting via `@Context`? I'm curious.